stl/inc/random

```diff
@@ -263,7 +288,38 @@ _NODISCARD _Real generate_canonical(_Gen& _Gx) { // build a floating-point value
     return _Ans / _Factor;
 }
 
-#define _NRAND(eng, resty) (_STD generate_canonical<resty, static_cast<size_t>(-1)>(eng))
+template <class _Real, class _Gen>
+_NODISCARD _Real _Nrand_impl(_Gen& _Gx) { // build a floating-point value from random sequence
+    _RNG_REQUIRE_REALTYPE(_Nrand_impl, _Real);
+
+    constexpr auto _Digits  = static_cast<size_t>(numeric_limits<_Real>::digits);
+    constexpr auto _Bits    = ~size_t{0};
+    constexpr auto _Minbits = _Digits < _Bits ? _Digits : _Bits;
+
+    if constexpr (_Select_invoke_traits<decltype(&_Gen::min)>::_Is_invocable::value
+                  && _Select_invoke_traits<decltype(&_Gen::max)>::_Is_invocable::value && _Minbits <= 64) {

```

Comment:
Although it would be pathological, there's nothing stopping a Standard URBG from having an *overloaded* `min()` and `max()` (e.g. overloading `min(meow)`), as long as zero-argument calls compile and meet the Standard's requirements. Therefore, writing `&_Gen::min` etc. could be ambiguous. I believe we'll need to use SFINAE to detect whether `min/max` are static. @CaseyCarter suggests using one of the Standard's requirements `bool_constant<(G::min() < G::max())>::value`, as that inspects both `min/max`. (I don't think we need to replicate the entire `uniform_random_bit_generator` concept in `void_t` SFINAE machinery, as we merely need to distinguish Standard from TR1-era tech.)